cpp/negentropy_wrapper.cpp

```diff
@@ -18,20 +23,27 @@ void printHexString(std::string_view toPrint){
 
 void* storage_new(const char* db_path, const char* name){
     negentropy::storage::BTreeMem* storage;
-/* 
-    auto env = lmdb::env::create();
-    env.set_max_dbs(64);
-    env.open(db_path, 0);
+    try {
+        /*
+        auto env = lmdb::env::create();
+        env.set_max_dbs(64);
+        env.open(db_path, 0);
 
-    lmdb::dbi btreeDbi;
+        lmdb::dbi btreeDbi;
 
-    {
-        auto txn = lmdb::txn::begin(env);
-        btreeDbi = negentropy::storage::BTreeMem::setupDB(txn, name);
-        txn.commit();
-    } */
+        {
+            auto txn = lmdb::txn::begin(env);
+            btreeDbi = negentropy::storage::BTreeMem::setupDB(txn, name);
+            txn.commit();
+        }
+        */
+
+        storage = new negentropy::storage::BTreeMem();
+    } catch (const std::exception& e) {
+        lastError = e.what();
```

Comment:
I think we can add some more context on each exception
```suggestion
        lastError = std::string("exception in storage_new: ") + e.what();
```

Reply:
> I think we can add some more context on each exception

Thanks! I'll also explore other possibilities similar to your suggestion.